### PR TITLE
feat: pool metadata task queue / single-tenant projection support

### PR DESCRIPTION
**Pull request by mkazlauskas**
# Context

1. We need to project stake pool metadata in a non-blocking way
2. We need to support projection variants, e.g. projecting all utxo (multi-tenant) vs projecting only utxo of provided set of addresses

# Proposed Solution

1. Project into [pg-boss](https://www.npmjs.com/package/pg-boss) jobs 1bf5e6e981f467913b5159c9e21d2f9e81a803c8...adbe6d632cac3fea17c236f8c609c7d3a1acd3bc
2. Refactor projection and projection-typeorm packages to remove a layer of abstraction (`{projections, sinks}`), which makes configuration more explicit (and more flexible as a result) 809f6956b62c2c18bd14f7ed11ed845a29ab940c...081713b32fa6fcd968ecb7c84183156c3f83b81e. Also, implement UTXO projection to demonstrate new capabilities: f237e873f70ff006e731e180e68a5d881d709e43...3ee7ba5211b8c8f3ff7c582b6246e6b094adeb09

# Important Changes Introduced

See commit log and an updated db schema diagram

schema

### CODE REVIEW

**Review by mirceahasegan — DISMISSED** [review]
Great job, @mkazlauskas ! 🧙‍♂️

File: packages/projection-typeorm/src/operators/storeUtxo.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                output: identifiers[producedIndex],
+                quantity
+              })
+            )
+        );
+        if (tokens.length > 0) {
+          await tokensRepository.insert(tokens);
+        }
+      }
+      // TODO: Review: maybe find a way to batch this query
```
Comment (rhyslbw) [inline packages/projection-typeorm/src/operators/storeUtxo.ts:48]: Log as a ticket for future improvement.

**Review by rhyslbw — APPROVED** [review]
Excellent work @mkazlauskas. Nice refactoring :rocket:

File: packages/projection-typeorm/src/operators/storeUtxo.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/projection-typeorm/src/operators/storeUtxo.ts:48]: ADP-2970

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**